rpcs3/Emu/RSX/Overlays/overlay_message.cpp

```diff
@@ -101,13 +191,49 @@ namespace rsx
 
 			compiled_resource cr{};
 
-			for (auto& item : m_queue)
+			for (auto& item : m_vis_items_top)
+			{
+				cr.add(item.get_compiled());
+			}
+
+			for (auto& item : m_vis_items_bottom)
 			{
 				cr.add(item.get_compiled());
 			}
 
 			return cr;
 		}
 
+		bool message::message_exists(message_pin_location location, localized_string_id id)
+		{
+			return message_exists(location, get_localized_u32string(id));
+		}
+
+		bool message::message_exists(message_pin_location location, const std::string& msg)
+		{
+			return message_exists(location, utf8_to_u32string(msg));
+		}
+
+		bool message::message_exists(message_pin_location location, const std::u32string& msg)
+		{
+			auto check_list = [&](const std::deque<message_item>& list)
+			{
+				return std::any_of(list.cbegin(), list.cend(), [&](const message_item& item)
+				{
+					return item.text_matches(msg);
+				});
+			};
+
+			switch (location)
+			{
+			case message_pin_location::top:
+				return check_list(m_ready_queue_top) || check_list(m_vis_items_top);
+			case message_pin_location::bottom:
+				return check_list(m_ready_queue_bottom) || check_list(m_vis_items_bottom);
+			default:
+				fmt::throw_exception("Unreachable");
```

Comment:
if you remove this we'll get a warning if we forget to add a new location here.

Reply:
![image](https://user-images.githubusercontent.com/15904127/216483288-2bf98c01-5d8c-4db1-8b3c-1a207822fe70.png)

Reply:
Added a return false, though should really throw. We don't enforce warnings as error so many warning go unnoticed.